src/transaction/creation_common.rs

```diff
@@ -0,0 +1,883 @@
+    ShowSimpleArgExamples,
+    ShowJsonArgExamples,
+    Verbose,
+    NodeAddress,
+    RpcId,
+    SpeculativeExec,
+    SecretKey,
+    Input,
+    Output,
+    Force,
+    TransferAmount,
+    TransferTargetAccount,
+    TransferId,
+    Timestamp,
+    Ttl,
+    ChainName,
+    SessionCode,
+    SessionArgSimple,
+    SessionArgsJson,
+    SessionArgsComplex,
+    SessionHash,
+    SessionName,
+    SessionPackageHash,
+    SessionPackageName,
+    SessionEntryPoint,
+    SessionVersion,
+    SessionTransfer,
+    SessionAccount,
+    PaymentAmmount,
+    PaymentMode,
+    }
+
+/// Handles providing the arg for and executing the show-simple-arg-examples option.
+pub(super) mod show_simple_arg_examples {
+    use super::*;
+
+    pub(in crate::transaction) const ARG_NAME: &str = "show-simple-arg-examples";
+    const ARG_ALIAS: &str = "show-arg-examples";
+    const ARG_SHORT: char = 'e';
+    const ARG_HELP: &str =
+        "If passed, all other options are ignored and a set of examples of session-/payment-args \
+        is printed";
+
+    pub(in crate::transaction) fn arg() -> Arg {
+        Arg::new(ARG_NAME)
+            .alias(ARG_ALIAS)
+            .long(ARG_NAME)
+            .short(ARG_SHORT)
+            .required(false)
+            .action(ArgAction::SetTrue)
+            .help(ARG_HELP)
+            .display_order(DisplayOrder::ShowSimpleArgExamples as usize)
+    }
+
+    pub(in crate::transaction) fn get(matches: &ArgMatches) -> bool {
+        if let Some(true) = matches.get_one::<bool>(ARG_NAME) {
+            println!("Examples for passing values via --session-arg or --payment-arg:");
+            println!("{}", simple_args_help::supported_cl_type_examples());
+            return true;
+        }
+
+        false
+    }
+}
+
+/// Handles providing the arg for and executing the show-json-arg-examples option.
+pub(super) mod show_json_args_examples {
+    use super::*;
+
+    pub(in crate::transaction) const ARG_NAME: &str = "show-json-args-examples";
+    const ARG_SHORT: char = 'j';
+    const ARG_HELP: &str = "If passed, all other options are ignored and a set of examples of \
+        session-/payment-args-json is printed";
+
+    pub(in crate::transaction) fn arg() -> Arg {
+        Arg::new(ARG_NAME)
+            .long(ARG_NAME)
+            .short(ARG_SHORT)
+            .required(false)
+            .action(ArgAction::SetTrue)
+            .help(ARG_HELP)
+            .display_order(DisplayOrder::ShowJsonArgExamples as usize)
+    }
+
+    pub(in crate::transaction) fn get(matches: &ArgMatches) -> bool {
+        if let Some(true) = matches.get_one::<bool>(ARG_NAME) {
+            println!("Examples for passing values via --session-args-json or --payment-args-json:");
+            println!();
+            println!("{}", json_args_help::info_and_examples());
+            return true;
+        }
+
+        false
+    }
+}
+
+pub(super) fn session_str_params(matches: &ArgMatches) -> SessionStrParams<'_> {
```

Comment:
We could maybe aim to move away from `SessionStrParams` for Transactions, leaving it only for Deploys?

In the same way we currently have `cli::transfer` not taking a general purpose `SessionStrParams`, I think it would be best to have separate functions for each of the `TransactionV1Builder`'s `new_xxx` methods, meaning none of them would need a `SessionStrParams`.